### PR TITLE
Update Go version to 1.18 

### LINKED ISSUE #262: Update Go version to 1.18
Goal: update the Go version used throughout the repository to Go 1.18.

AC:
- Update the root go.mod
- Update the GHA workflows to use Go 1.18
- Update any Makefile/script references to Go 1.17

### PR DESCRIPTION
**Pull request by EdwinWalela**
Closes #262 

Running `make lint` produces the following output
```bash
WARN [linters context] gosimple is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] staticcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] stylecheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] tparallel is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] unparam is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649. 
WARN [linters context] unused is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
```

### CODE REVIEW

**Review by timflannagan — APPROVED** [review]
Thanks - these changes look right to me. Can we add the "Closes #262" in the PR description, or update the "Development" section on the righthand side, and link the bump Go 1.18 issue so the github automation will close that issue once these changes merge?

Also: should we create a follow-up issue that bumps the golangci-lint version once those temporarily disabled linters have been re-enabled?

File: hack/tools/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/operator-framework/rukpak/hack/tools
 
-go 1.17
+go 1.18
 
 require (
 	github.com/golangci/golangci-lint v1.45.1
```
Comment (joelanford) [inline hack/tools/go.mod:6]: Bump this version to match the version we're using in github actions:
```suggestion
	github.com/golangci/golangci-lint v1.45.2
```

**Review by joelanford — COMMENTED** [review]
Just one small thing I noticed, but looks great! Thanks!

File: hack/tools/go.mod (hunk 1 above)
Comment (EdwinWalela) [reply hack/tools/go.mod]: @joelanford Seems I missed out on that. I've pushed the update

**Comment by EdwinWalela** [comment]
@timflannagan  I've updated the description as requested. 

> should we create a follow-up issue that bumps the golangci-lint version once those temporarily disabled linters have been re-enabled?

Sounds good, does that mean we should remain at `v1.45.1` in the meantime?

File: hack/tools/go.mod [hunk 2]
```diff
@@ -1,9 +1,9 @@
 module github.com/operator-framework/rukpak/hack/tools
 
-go 1.17
+go 1.18
 
 require (
-	github.com/golangci/golangci-lint v1.45.1
+	github.com/golangci/golangci-lint v1.45.2
 	github.com/goreleaser/goreleaser v1.6.1
```
Comment (timflannagan) [inline hack/tools/go.mod:7]: Do we need to also bump goreleaser to a version that supports Go 1.18? Looking at their [releases](https://github.com/goreleaser/goreleaser/releases), it seems like the v1.7.0+ tags support Go 1.18. The release workflow, which is responsible for running goreleaser under-the-hood appears to have passed, so we might be fine as long as the root go.mod specified as Go 1.18 version? Thoughts:?

**Comment by timflannagan** [comment]
@EdwinWalela I think I'm fine with sticking with that v1.45.2 golangci-lint release, as it looks like v1.45.1 also contains the disabled linters.

File: hack/tools/go.mod (hunk 2 above)
Comment (EdwinWalela) [reply hack/tools/go.mod:7]: goreleaser 1.8.3 is on go 1.18 too. We can bump it to the latest (1.8.3) too

**Review by exdx — APPROVED** [review]

File: hack/tools/go.mod (hunk 2 above)
Comment (joelanford) [reply hack/tools/go.mod:7]: I think we also need to re-run `go mod tidy` in `hack/tools`. I'd expect to see some `hack/tools/go.sum` changes show up.

Comment (EdwinWalela) [reply hack/tools/go.mod]: I've bumped goreleaser to 1.8.3 and re-ran `go mod tidy` in `hack/tools`

Comment (EdwinWalela) [reply hack/tools/go.mod]: I've bumped goreleaser to 1.8.3 and re-ran `go mod tidy` in `hack/tools`

**Review by timflannagan — COMMENTED** [review]
Can we rebase against the latest commits in the main branch? Joe had updated the `tide` Makefile target to also check for nested go.sum changes in #278. GHA has some annoying behavior around PRs testing against a snapshot of the main branch when a PR has been opened, vs. the current snapshot of the main branch, so I just want to triple check we're not regressing main for that verify check.

Should we also fixup the latest commits into a single commit, e.g. commit for updating the Go version, and another commit for bumping any of the tooling dependencies?

**Comment by timflannagan** [comment]
(apologies for the review churn btw)

**Comment by timflannagan** [comment]
@EdwinWalela It looks like we should be all set for rebasing looking at the [sanity check output](https://github.com/operator-framework/rukpak/runs/6219207726?check_suite_focus=true). You may need to run `make tidy` locally and commit the hack/tools/go.sum changes to make that check happy. I'm cool with squashing too.

**Comment by EdwinWalela** [comment]
@timflannagan I've commit the `hack/tools/go.sum` changes and squashed all the commits into one. Let me know if there's something I've missed out

**Review by timflannagan — APPROVED** [review]

**Pull request merged**